### DISCUSSION TITLE
Joiner family parameter names

### DISCUSSION THREAD

**Discussion by Vincent-Maladiere** [skrub-data/skrub#751] [Polls]
### Parameters names

During the skrub meeting on 25/09, we discussed bringing more consistency to the parameter names of Joiner-like transformers. The current conclusions are to match pandas API `pd.merge`  with `fuzzy_join` function to ease adoption while differentiating from pandas for the transformer classes.

This table recaps the current consensus:

| name                          | fuzzy_join        | Joiner            | InterpolateJoiner | AggJoiner         |
|-------------------------------|-------------------|-------------------|-------------------|-------------------|
| auxiliary dataframe parameter | left, right       | aux_table         | aux_table         | aux_table         |
| keys parameter                | left_on, right_on | main_key, aux_key | main_key, aux_key | main_key, aux_key |
| suffix parameter              | suffixes          | suffix            | suffix            | suffix            |
| match_score parameter         | match_score       | match_score       | NA                | NA                |

### Multiple tables classes

We also agreed on only using a single auxiliary table in `Joiner`, `InterpolateJoiner`, and `AggJoiner`. The reason is to make these API easier to use. For each object, we will create a `Multi` version to deal with multiple tables, respectively `MultiJoiner`, `MultiInterpolateJoiner`, `MultiAggJoiner`. These objects will have the same parameters as their single table version.

### Classes names

Also, we voted on naming classes either verbs (e.g. `Join`) or nouns (e.g. `Joiner`) throughout all skrub API. The vote leaned on verbs rather than nouns (5 against 4). We can use this poll to continue this discussion and change our votes if we want to. 

cc @jovan-stojanovic, @jeromedockes, @myungkim930, @rcap107, @GaelVaroquaux, @flefebv, @LeoGrin, @tigerchen52

**Comment by jeromedockes** [comment]
as mentioned in [742](https://github.com/skrub-data/skrub/pull/742#discussion_r1339905687) we can add the `key` parameter which can be use in place of `main_key` and `aux_key` when they are the same, similar to `on` in pandas and polars or `using` in sql
